server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedNumericDVIndexFieldData.java

```diff
@@ -53,21 +59,58 @@
 import java.io.IOException;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Objects;
 import java.util.function.LongUnaryOperator;
 
 /**
  * FieldData backed by {@link LeafReader#getSortedNumericDocValues(String)}
  * @see DocValuesType#SORTED_NUMERIC
  */
-public class SortedNumericDVIndexFieldData extends DocValuesIndexFieldData implements IndexNumericFieldData {
-    private final NumericType numericType;
+public class SortedNumericDVIndexFieldData implements IndexNumericFieldData {
+    public static class Builder implements IndexFieldData.Builder {
+
+        private NumericType numericType;
+        public Builder numericType(NumericType type) {
```

Comment:
Is `numericType` required? Maybe it can be `final` a ctor arg?

Reply:
I wonder if *all* of these sorts of args can be ctor args.

Reply:
Yeah, I think they can.  I'll take a pass at it.